### PR TITLE
fix(internal): fix self-signed detection with scopes

### PR DESCRIPTION
**Pull request by codyoss**
Removed the requirement on length of scopes passed. Checking the EnableJwtWithScope internal option should be sufficient since it will only be passed with our generated clients that will always include at least the platform scope and a default audience. 
Fixes: #1092

### CODE REVIEW

**Comment by codyoss** [comment]
cc @shinfan

File: internal/creds.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func credentialsFromJSON(ctx context.Context, data []byte, ds *DialSettings) (*g
 }
 
 func isSelfSignedJWTFlow(data []byte, ds *DialSettings) (bool, error) {
-	if (ds.EnableJwtWithScope || ds.HasCustomAudience() || len(ds.GetScopes()) == 0) &&
+	if ((ds.EnableJwtWithScope && len(ds.GetScopes()) > 0) || ds.HasCustomAudience()) &&
 		ds.ImpersonationConfig == nil {
```
Comment (tbpg) [inline internal/creds.go:103]: Refactor these two lines to get more descriptive names and avoid lots of logical nesting?

File: internal/creds.go [hunk 2]
```diff
@@ -99,7 +99,7 @@ func credentialsFromJSON(ctx context.Context, data []byte, ds *DialSettings) (*g
 }
 
 func isSelfSignedJWTFlow(data []byte, ds *DialSettings) (bool, error) {
-	if (ds.EnableJwtWithScope || ds.HasCustomAudience() || len(ds.GetScopes()) == 0) &&
+	if ((ds.EnableJwtWithScope && len(ds.GetScopes()) > 0) || ds.HasCustomAudience()) &&
```
Comment (shinfan) [inline internal/creds.go:102]: I don't feel we need to check for the scope here when scope is enabled for JWT

1) If scope is non-empty, JWT is used
2) If scope is empty, audience must be non-empty, in this case JWT is used too.

Does this make sense?

Comment (codyoss) [reply internal/creds.go]: I agree, will remove as that is another way to fix this.

File: internal/creds.go (hunk 1 above)
Comment (codyoss) [reply internal/creds.go]: Done

**Review by tbpg — APPROVED** [review]

**Pull request merged**